ospd/xml.py

```diff
@@ -111,3 +111,56 @@ def get_elements_from_dict(data: Dict[str, Any]) -> List[Element]:
         responses.append(elem)
 
     return responses
+
+
+class XmlStringHelper:
+    """ Class with methods to help the creation of a xml object in
+    string format.
+    """
+
+    def create_element(self, elem_name: str, end: bool = False) -> bytes:
+        """ Get a name and create the open element of an entity."""
+
+        if end:
+            ret = "</%s>" % elem_name
```

Comment:
I don't think we should create xml elements via string manipulation even if it uses less memory. We should still use the element tree api and yield converted data instead.

```python
xml = Element(elem_name)
return et.tostring(xml)
```
I want to avoid to introduce nasty xml injections. Therefore it is better to use a xml api for creating the element strings.

Reply:
As we have to iterate the vts dictionary in the middle, we have to send this by pieces. Otherwise, we have to build the huge xml element before starting to send the data, and this consumes too much memory.